src/flask/app.py

```diff
@@ -2146,11 +2150,11 @@ def create_url_adapter(self, request):
             # If subdomain matching is disabled (the default), use the
             # default subdomain in all cases. This should be the default
             # in Werkzeug but it currently does not have that feature.
-            subdomain = (
-                (self.url_map.default_subdomain or None)
-                if not self.subdomain_matching
-                else None
-            )
+            if self.subdomain_matching:
+                subdomain = None
+            else:
+                subdomain = self.url_map.default_subdomain or None
+
```

Comment:
Why not
`subdomain = None if self.subdomain_matching else self.url_map.default_subdomain`

Reply:
That would not convert `''` to `None` as the current code does.

Reply:
@sukurcf Because it's will too long (more than 78 characters), that's why.